### PR TITLE
[DDW-518] Tile view of stake pools ranked by rewards

### PR DESCRIPTION
**Pull request by aleksandardjordjeviciohk**
This PR updates Stake pools tile view with tiles ranked by rewards.

## User story: 

As a Daedalus user, I want to be able to see the pools in a user-friendly tile view, ranked descending according to potential reward yield so that I can have a complete overview and make potential educated decisions.

## Scenario: Viewing the tile view in the Delegation Center

- Given that I am in Daedalus stake pools page
- When I click “Rewards tile view” icon
- Then I am shown a tile view of the stake pools 
- And they are ranked in a descending order according to potential reward yield

## Todos

- [x] Add sorting/ordering button to search toolbar of stake pools tile view
- [x] Add ranking number to tiles
- [x] Implement ordering by ranking
- [x] Update storybook for stake pools tile view
- [x] Display - instead of 0 ada when estimated rewards are 0
- [x] Make this Flight and Testnet only feature (configurable)

## Screenshots

Screenshot 2021-03-05 at 19 17 56
Screenshot 2021-03-05 at 19 18 14
Screenshot 2021-03-05 at 19 18 04
Screenshot 2021-03-05 at 19 18 16

---

## Testing Checklist

- [Slack QA thread](https://input-output-rnd.slack.com/archives/GGKFXSKC6/p1614770061158100)

---

## Test Cases
```gherkin
Scenario 1: Viewing the tile view in the Delegation Center
Given that I am in Daedalus stake pools page
When I click “Rewards tile view” icon
Then I am shown a tile view of the stake pools 
And they are ranked in a descending order according to potential reward yield

Scenario 2. Choosing rewards tile view when no wallets are loaded in Daedalus
Given that I am on Daedalus
And the chain folder is fully synced 
And I don’t have any wallets loaded
And I navigate to the Stake Pool Tab
And I click on the new Rewards Tile View Icon
When I change the value for how much I intend to delegate on the slider
Then the stake pools will be reordered to match my selection from highest to lowest
```
### Basics

- [ ] PR has been assigned and has appropriate labels (`feature`/`bug`/`chore`, `release-x.x.x`)
- [ ] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [ ] PR has a good description that summarizes all changes
- [ ] PR has default-sized Daedalus window screenshots or animated GIFs of important UI changes:
  - [ ] In English
  - [ ] In Japanese
- [ ] CHANGELOG entry has been added to the top of the appropriate section (*Features*, *Fixes*, *Chores*) and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance and unit tests are passing (`yarn test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn package` / CI builds)
- [ ] There are no *flow* errors or warnings (`yarn flow:test`)
- [ ] There are no *lint* errors or warnings (`yarn lint`)
- [ ] There are no *prettier* errors or warnings (`yarn prettier:check`)
- [ ] There are no missing translations (running `yarn manage:translations` produces no changes)
- [ ] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [ ] Japanese text changes are proofread and approved (Junko Oda)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`yarn storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [ ] Important parts of the code are properly commented and documented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-renderings
- [ ] Any code that only works in main process is neatly separated from components

### Testing
- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review
- [ ] Merge the PR
- [ ] Delete the source branch
- [ ] Move the ticket to `done` column on the YouTrack board
- [ ] Update Slack QA thread by marking it with a green checkmark

### CODE REVIEW

**Review by ManusMcCole — APPROVED** [review]
Looks great @DeeJayElly 👍

**Review by miorsufianiohk — APPROVED** [review]
Apart from my suggestion on the need for tooltips, LGTM. Well done @aleksandardjordjeviciohk 👍

**Comment by gabriela-ponce** [comment]
@aleksandardjordjeviciohk I found to small issues on build 16910:

- The X button on the search bar is not working properly. Check [this video.](https://drive.google.com/file/d/1w5mi2p8I0j6_MIJ6rD2h0uo2qINJ9ulj/view?usp=sharing)
- Also, it overlaps the divider. Check [this screenshot](https://drive.google.com/file/d/1Yb5qz_K95rW-rI3FEVyukhnkXGCj9skz/view?usp=sharing) on another theme where is more visible.

**Comment by aleksandardjordjeviciohk** [comment]
> @aleksandardjordjeviciohk I found two small issues on build 16910:
> 
> * The X button on the search bar is not working properly. Check [this video.](https://drive.google.com/file/d/1w5mi2p8I0j6_MIJ6rD2h0uo2qINJ9ulj/view?usp=sharing)
> * Also, it overlaps the divider. Check [this screenshot](https://drive.google.com/file/d/1Yb5qz_K95rW-rI3FEVyukhnkXGCj9skz/view?usp=sharing) on another theme where is more visible.

@gabriela-ponce can you recheck now, both of these problems are fixed

**Comment by gabriela-ponce** [comment]
@aleksandardjordjeviciohk The mentioned issues are fixed. However, on the new tooltips, the translation is missing for the "clear" button. Check [this screenshot](https://drive.google.com/file/d/11_CNFQTu4glE3Zes3tLt3S0s9TpFxWUn/view?usp=sharing).

**Comment by nikolaglumac** [comment]
> @aleksandardjordjeviciohk The mentioned issues are fixed. However, on the new tooltips, the translation is missing for the "clear" button. Check [this screenshot](https://drive.google.com/file/d/11_CNFQTu4glE3Zes3tLt3S0s9TpFxWUn/view?usp=sharing).

@aleksandardjordjeviciohk please fix this 🙏 

We already have translation for this here: https://github.com/input-output-hk/daedalus/blob/develop/source/renderer/app/i18n/locales/ja-JP.json#L929

**Comment by aleksandardjordjeviciohk** [comment]
> > @aleksandardjordjeviciohk The mentioned issues are fixed. However, on the new tooltips, the translation is missing for the "clear" button. Check [this screenshot](https://drive.google.com/file/d/11_CNFQTu4glE3Zes3tLt3S0s9TpFxWUn/view?usp=sharing).
> 
> @aleksandardjordjeviciohk please fix this 🙏
> 
> We already have translation for this here: https://github.com/input-output-hk/daedalus/blob/develop/source/renderer/app/i18n/locales/ja-JP.json#L929

@gabriela-ponce @nikolaglumac 
It was wrongly referenced in the translations file, i have updated it with correct translation. Tnx for pointing it out.

**Comment by nikolaglumac** [comment]
@yakovkaravelov since @daniloprates is off for the rest of the week, I hope you can help us reviewing this one 🙏

**Comment by nikolaglumac** [comment]
@darko-mijic @gabriela-ponce please review this one too 🙏

**Comment by nikolaglumac** [comment]
@alexander-rukin please review 🙏

**Comment by alexander-rukin** [comment]
@aleksandardjordjeviciohk here we miss variable for "ADA symbol"  
image

File: source/renderer/app/components/staking/widgets/ThumbPoolContent.js [hunk 1]
```diff
@@ -44,22 +51,29 @@ export default class ThumbPoolContent extends Component<Props> {
...
-              <>
-                {numberOfRankedStakePools + 1}
-                <sup>*</sup>
-              </>
-            )}
-          </div>
-        ) : (
-          <div className={styles.noDataDash}>
-            <SVGInline svg={noDataDashBigImage} />
+        {isGridRewardsView && (
```
Comment (yakovkaravelov) [inline source/renderer/app/components/staking/widgets/ThumbPoolContent.js:54]: What about using following format?
```{isGridRewardsView ? () : ()}```

Comment (aleksandardjordjeviciohk) [reply source/renderer/app/components/staking/widgets/ThumbPoolContent.js:54]: @yakovkaravelov then we will have nested ternary operators warning, linter won't allow that.

Screen Shot 2021-03-04 at 4 28 56 AM

**Comment by darko-mijic** [comment]
I have added one todo.

Also, ada symbol is not visible in the Flight theme
image

**Comment by alexander-rukin** [comment]
> I have added one todo.
> 
> Also, ada symbol is not visible in the Flight theme
> image

because of https://github.com/input-output-hk/daedalus/pull/2426#issuecomment-790490390

**Comment by aleksandardjordjeviciohk** [comment]
> @aleksandardjordjeviciohk here we miss variable for "ADA symbol" fill color
> 
> image
> 
> stroke is hardcoded
> 
> image

@alexander-rukin fixed

**Comment by aleksandardjordjeviciohk** [comment]
> I have added one todo.
> 
> Also, ada symbol is not visible in the Flight theme
> image

@darko-mijic fixed

**Comment by alexander-rukin** [comment]
> @alexander-rukin fixed

not really, you have double opacities: 

image


image

**Comment by gabriela-ponce** [comment]
@aleksandardjordjeviciohk The tooltips look good now 👍 
There is a new issue, though: when clicking the "Grid rewards view" the stake pools and the buttons on the search bar disappear. Check [this](https://drive.google.com/file/d/1Pk1AASrSl15YPDutxYxzUfar6IA6vLic/view?usp=sharing) video. There is no error on the console.

**Comment by alexander-rukin** [comment]
image

shouldn't we have rewards here too instead of rank? cc @darko-mijic 
If possible I'd insist on it))

**Comment by aleksandardjordjeviciohk** [comment]
> @aleksandardjordjeviciohk The tooltips look good now 👍
> There is a new issue, though: when clicking the "Grid rewards view" the stake pools and the buttons on the search bar disappear. Check [this](https://drive.google.com/file/d/1Pk1AASrSl15YPDutxYxzUfar6IA6vLic/view?usp=sharing) video. There is no error on the console.

@gabriela-ponce i have added a small updated which covers this part. We shouldn't be able to see ranking grid view button on mainnet for now.

**Comment by aleksandardjordjeviciohk** [comment]
> image
> 
> shouldn't we have rewards here too instead of rank? cc @darko-mijic
> If possible I'd insist on it))

added

File: source/renderer/app/components/staking/stake-pools/StakePoolsSearch.js [hunk 2]
```diff
@@ -64,6 +64,7 @@ type Props = {
   onGridRewardsView?: Function,
   onListView?: Function,
   search: string,
+  isMainnet?: boolean,
```
Comment (nikolaglumac) [inline source/renderer/app/components/staking/stake-pools/StakePoolsSearch.js:67]: @aleksandardjordjeviciohk please make this required property

Comment (aleksandardjordjeviciohk) [reply source/renderer/app/components/staking/stake-pools/StakePoolsSearch.js]: updated

**Comment by nikolaglumac** [comment]
@aleksandardjordjeviciohk please fix flow errors.

**Comment by alexander-rukin** [comment]
Is it hard to implement https://input-output-rnd.slack.com/archives/G0186R8BWNR/p1614880043032500 in scope of this PR? 
I would be soooooo happy 

p.s. visual part of current task is already seems OK, so can approve in case of rejection

**Review by alexander-rukin — APPROVED** [review]

**Comment by darko-mijic** [comment]
The new view is not available in the latest Flight build. 
image

**Comment by nikolaglumac** [comment]
@darko-mijic all issues have been fixed now. Please review again 🙏

**Comment by gabriela-ponce** [comment]
@aleksandardjordjeviciohk I checked again [this issue](https://github.com/input-output-hk/daedalus/pull/2426#issuecomment-789756667) about the "clear" button and it still reproduces in grid and tile view (no longer reproduces on the new tile view). Check [this](https://drive.google.com/file/d/1Wb0ym-EUGRogTY66dJ2yxt3yFjld27vz/view?usp=sharing) video.

**Comment by miorsufianiohk** [comment]
There's a pipe symbol, on the very left hand side of the buttons when 'Reward view' is chosen which is inconsistent with other views. Tested on 17020
List view
List view
Reward view
Reward view
Tile view
Tile view

**Comment by nikolaglumac** [comment]
> There's a pipe symbol, on the very left hand side of the buttons when 'Reward view' is chosen which is inconsistent with other views. Tested on 17020
> List view
> List view
> Reward view
> Reward view
> Tile view
> Tile view

I believe the rewards view (having the pipe) is the correct one. 
@alexander-rukin can you please confirm?

**Comment by nikolaglumac** [comment]
@miorsufianiohk @gabriela-ponce I have fixed all the remaining issues. Please quickly check only those fixes and approve 🙏

**Review by gabriela-ponce — APPROVED** [review]

**Review by ManusMcCole — APPROVED** [review]

**Comment by alexander-rukin** [comment]
> > There's a pipe symbol, on the very left hand side of the buttons when 'Reward view' is chosen which is inconsistent with other views. Tested on 17020
> > List view
> > List view
> > Reward view
> > Reward view
> > Tile view
> > Tile view
> 
> I believe the rewards view (having the pipe) is the correct one.
> @alexander-rukin can you please confirm?

Yes -> The rewards view (having the pipe) is the correct one.

**Pull request merged**